apps/mobile/app/register.tsx

```diff
@@ -0,0 +1,54 @@
+import { router } from "expo-router";
+import { Text, View } from "react-native";
+
+import { useAuth } from "../stores/auth";
+import { trpc } from "../trpc/client";
+import { useForm } from "react-hook-form";
+import { CredentialsData } from "backend/auth/types";
+import { Button } from "ui/Button";
+import { ControlledTextInput } from "ui/ControlledTextInput";
+
+type RegisterData = CredentialsData & {
+  retypePassword: string;
+};
+
+export default function Register() {
+  const mutation = trpc.register.useMutation();
+  const setTokens = useAuth((state) => state.setTokens);
+  const { control, handleSubmit } = useForm<RegisterData>();
+
+  const onSubmit = async ({
+    email,
+    password,
+    retypePassword,
+  }: RegisterData) => {
+    console.log(email, password, retypePassword);
+    if (password !== retypePassword) {
+      console.log("Passwords must be the same!");
+      return;
+    }
+    const data = await mutation.mutateAsync({ email, password });
+    setTokens(data);
+    router.replace("/");
+  };
+
+  return (
+    <View style={{ flex: 1, justifyContent: "center", alignItems: "center" }}>
```

Comment:
Do we want anything like styled-components or tailwind?

Reply:
Nah, I'll use StyleSheet api, those inline styles are just for testing purposes
However, I'll add react-native-paper